packages/wp-now/src/config/providers/WpEnvConfigProvider.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs';
+import path from 'path';
+import { CliOptions } from './CliConfigProvider';
+import { WPEnvOptions } from './ConfigProvider';
+import { ConfigProviderInterface, WPNowOptions } from './ConfigProvider';
+
+const WP_ENV_FILE = '.wp-env.json';
+const WP_ENV_OVERRIDE_FILE = '.wp-env.override.json';
+
+const WP_ENV_TO_WP_NOW_MAPPER = {
+	phpVersion: 'phpVersion',
+	port: 'port',
+};
+
+class WpEnvConfigProvider implements ConfigProviderInterface {
+	priority: 1;
+
+	#options: CliOptions | null;
+	constructor(options: CliOptions) {
+		this.#options = options;
+	}
+
+	#mergeConfigs(
+		wpEnvConfig: WPEnvOptions,
+		wpOverrideEnvConfig: WPEnvOptions
+	): WPNowOptions {
+		if (!wpEnvConfig && !wpOverrideEnvConfig) {
+			return null;
+		}
+
+		if (!wpEnvConfig && wpOverrideEnvConfig) {
+			return wpOverrideEnvConfig;
+		}
+
+		const config: WPNowOptions = {};
+		for (const [key, value] of Object.entries(wpEnvConfig || {})) {
+			if (wpOverrideEnvConfig && wpOverrideEnvConfig[key]) {
+				config[key] = wpOverrideEnvConfig[key];
+				continue;
+			}
+			config[key] = value;
+		}
+		return config;
+	}
+
+	#getConfigPath(configName: string): string {
+		return path.join(
+			(this.#options.path as string) || process.cwd(),
+			configName
+		);
+	}
+
+	configExists(): boolean {
+		const wpEnvConfigPath = this.#getConfigPath(WP_ENV_FILE);
+		const wpEnvOverrideConfigPath =
+			this.#getConfigPath(WP_ENV_OVERRIDE_FILE);
+		return (
+			fs.existsSync(wpEnvConfigPath) ||
+			fs.existsSync(wpEnvOverrideConfigPath)
+		);
+	}
+
+	#toWpNow(wpenvConfig: WPEnvOptions): WPNowOptions {
+		if (!wpenvConfig) {
+			return null;
+		}
+		const config: WPNowOptions = {};
+		for (const [key, value] of Object.entries(wpenvConfig)) {
+			if (WP_ENV_TO_WP_NOW_MAPPER[key]) {
+				config[WP_ENV_TO_WP_NOW_MAPPER[key]] = value;
+			}
+		}
+		return config;
```

Comment:
What do you think about this formulation? It's shorter, keeps everything you need to think about in the same place, and it will be easy to extend to nested properties:

```suggestion
		return {
		    phpVersion: wpenvConfig.phpVersion,
		    port: wpenvConfig.port
		};
```

Reply:
Agree. It's smaller and does the work. Thanks for your proposal!